src/driver/compile.ml

```diff
@@ -222,16 +231,50 @@ let link : compiled list -> _ =
         (match c.m with
         | Module _ -> Atomic.incr Stats.stats.linked_units
         | Mld _ -> Atomic.incr Stats.stats.linked_mlds);
-        { output_file = Fpath.(set_ext "odocl" c.output_file); src = None }
+        {
+          output_file = Fpath.(set_ext "odocl" c.output_file);
+          src = None;
+          pkgname = c.pkgname;
+        }
         :: impl
   in
   Fiber.List.map link compiled |> List.concat
 
+let index_one output_dir pkgname _pkg =
+  let dir = Fpath.(output_dir / pkgname) in
+  let dst = Fpath.(dir / "index.odoc-index") in
+  let include_rec = Fpath.Set.singleton dir in
+  Odoc.compile_index ~json:false ~dst ~include_rec ()
+let index odoc_dir pkgs = Util.StringMap.iter (index_one odoc_dir) pkgs
+
+let sherlodoc_index_one ~html_dir ~odoc_dir pkgname _pkg_content =
+  ignore @@ Bos.OS.Dir.create Fpath.(html_dir / pkgname);
+  let format = `js in
+  let inputs = [ Fpath.(odoc_dir / pkgname / "index.odoc-index") ] in
+  let dst = Fpath.(html_dir / pkgname / "sherlodoc_db.js") in
```

Comment:
This filename is hardcoded and reused. It would be better to return it or something like that.